### ISSUE TITLE
[Feature Request] Customizable hotkeys for Save States and Load States

### ISSUE THREAD

**Issue by ChronoCoins** [Arisotura/melonDS#1183]
I'm not really a big fan of holding the shift key and pressing an "F Key" to make save states. (Shift + F key)

Most emulators, including DeSmuME, would let me customize the hotkeys for making saves and loading savestates.

(Here's a photo of DeSmuME's hotkey config menu for savestates with my preferred keys.)

![DeSmuME-VS2019-x64-Release_2021-08-19_15-42-14](https://user-images.githubusercontent.com/60282457/130141931-8d85975d-a88a-4df4-8e24-37411f209812.png)

**Comment by BluePlasticCup** [comment]
This would be a godsend.
It's one of those small but important parts of emulation that you don't fully appreciate until they're missing.

**Comment by ReaderGuy42** [comment]
Have there been any updates on this?

**Comment by Pedsouji** [comment]
Any updates/progress/workaround on this? I'm currently using AYANEO 2 (handheld console) and it's my biggest gripe to not have custom hotkey for save/load state, as well as fast forward toggle (currently, you need to hold the key). I can imagine Steam Deck owners would have the same sentiments as mine.

**Comment by Squ1dCake** [comment]
I am also hoping for this as a Steam Deck owner.

**Comment by RPGShooter17020** [comment]
> I am also hoping for this as a Steam Deck owner.

For SteamDeck you can click the Steam button and go to controller settings > edit layout > under a button add a command for shift and then add a sub command for F1(or any other save state slot) which will allow you to create a save state. Then just add a  command for another button for the corresponding F# key to load said save state. If you run a game through a separate launcher(as in through say a Heart Gold icon in the library and not the MelonDS icon) for it to work.

![20230811_190309](https://github.com/melonDS-emu/melonDS/assets/77864633/8debcb77-1fb1-469a-9ecb-99154034d03d)

**Comment by Shideravan** [comment]
Why are the Melon DS devs so reluctant to include functionality to allow this?

**Comment by Jaklyy** [comment]
Focus has been elsewhere.
Its an easy change, but if everyone capable of implementing it is focusing on other priorities then it wont get done.

**Comment by Clock-name** [comment]
Yeah but, for like 4-5 years is kinda silly, it's the only reason i use other emulators half the time because i backup my saves with the savestate system, so im always 1 tap away from losing much progress. I am once again campaigning to add such a feature because this is the best emulator i've seen but it's got such a weird sometimes invasive problem.

**Comment by poudink** [comment]
you can undo state load with f12 so there isn't a risk of losing progress

but also you probably shouldn't use savestates to back up anything. better to just backup your save folder.

**Comment by xXRaptorSc0pezXx** [comment]
I am also hoping for this feature! Honestly this is just a basic emulator feature, this should have already been added and really needs to be.

**Comment by Katknapp** [comment]
Adding a +1 to this, kind of flow-breaking to have to change to keyboard to savestate.
